docs/docs/development/documentation.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Documentation
+sidebar_label: Documentation
+---
+
+This document outlines how to test your documentation changes locally and prepare the changes for a pull request.
+
+The documentation is written with [Docusaurus](https://docusaurus.io/). The ZMK source code has all of the necessary Docusaurus dependencies included but referencing their documentation can be helpful at times.
+
+The general process for updating the ZMK documentation is:
+
+1. Update the documentation
+2. Test the changes locally
+3. Ensure the sources are formatted properly and linted
+4. Create a Pull Request for review and inclusion into the ZMK sources
+
+:::note
+If you are working with the documentation from within VS Code+Docker please be aware the documentation will not be auto-generated when making changes while the server is running. You'll need to restart the server when saving changes to the documentation.
+:::
+
+## Preparation
+
+After making updates to the documentation locally, you'll want to test the updates locally using the following procedure. The `npm` commands and first step will need to be run from a terminal.
+
+## Testing Documentation Updates Locally
+
+To verify documentation updates locally, follow the following procedure. The `npm` commands and first step will need to be run from a terminal.
+
+1. Navigate to the `docs` folder
+2. Run `npm ci` to build the necessary tools if you haven't run it before or the ZMK sources were updated
+3. Run `npm start` to start the local server that will let you see your documentation updates via a web browser
+4. If a web browser doesn't open automatically: you'll need to open a browser window or tab and navigate to `http://localhost:3000` to view your changes
+5. Verify the changes look good
+6. Run the formatting and checks from the section below
+7. Submit any changes as a Pull Request
+
+## Formatting and Linting Your Changes
+
+Prior to submitting a documentation pull request, you'll want to run the format and check commands. These commands are run as part of the verification process on pull requests so it's good to run them ahead of submitting documentation updates.
+
+The format command can be run with the following procedure in a terminal that's inside the ZMK source directory.
+
+1. Navigate to the `docs` folder
+2. Run `npm run prettier:format`
+
+The check commands can be run with the following procedure in a terminal that's inside the ZMK source directory.
+
+1. Navigate to the `docs` folder
+2. Run `npm run prettier:check`
+3. Run `npm run lint`
+4. Run `npm run build`
+
+:::warning
+If any of the above steps throw an error, they need to be addressed and all of the checks re-run prior to submitting a pull request.
+:::
```

Comment:
Maybe after this you could make a small section about making a pull request with a link to some resource that would help someone new to pull requests on GitHub?

Reply:
I would like to leave the PR creation piece off these docs. There was recent discussion in discord about PR's and generally folks were linking to the github documentation.

If we do anything related to Pull Requests I think it should be a dedicated area within the documentation and cross linked from other pages where relevant.

Reply:
I meant more of a section just saying to create a pull request after the formatting and linting are finished. It doesn't need to be in depth or anything, more of a resolution to the page, which states creating the pull request as the last step.

Reply:
Will add and update